HCTextField/ViewController.swift

```diff
@@ -8,16 +8,103 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, HCTextFieldDelegate {
+
+    @IBOutlet weak var nameField: HCTextField!
+    @IBOutlet weak var emailField: HCTextField!
+    @IBOutlet weak var passwordField: HCTextField!
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+
+        nameField.setCheckType(.empty, errorMessage: "cann't be empty")
+        nameField.textFieldDelegate = self
+
+        emailField.setCheckType(.email, errorMessage: "not a valid email")
+        emailField.textFieldDelegate = self
+
+        passwordField.setCheckType(.length, errorMessage: "6 ~ 20 characters", minLength: 6, maxLength: 20)
+        passwordField.textFieldDelegate = self
+
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
 
+    // MARK: - Actions
+
+    @IBAction func loginButtonTapped() {
+        _ = passwordField.resignFirstResponder()
+
+        var title = ""
+        var message: String?
+
+        if nameField.passedCheck && emailField.passedCheck && passwordField.passedCheck {
```

Comment:
we need to be able to simplify this. instead of using chained "&&" there should be one overall check to perform all checks.